packages/material-ui/src/Select/Select.test.js

```diff
@@ -16,34 +15,39 @@ import Input from '@material-ui/core/Input';
 import InputLabel from '@material-ui/core/InputLabel';
 import Select from '@material-ui/core/Select';
 import Divider from '@material-ui/core/Divider';
+import classes from './selectClasses';
 
 describe('<Select />', () => {
-  let classes;
   const mount = createMount();
   // StrictModeViolation: triggers "not wrapped in act()" warnings from timers.
   const render = createClientRender({ strict: false });
 
-  before(() => {
-    classes = getClasses(<Select />);
-  });
-
-  describeConformance(<Select value="" />, () => ({
+  describeConformanceV5(<Select value="" />, () => ({
     classes,
     inheritComponent: Input,
+    render,
     mount,
     refInstanceof: window.HTMLDivElement,
-    skip: ['componentProp', 'rootClass'],
+    muiName: 'MuiSelect',
+    skip: ['componentProp', 'componentsProp', 'rootClass', 'themeVariants', 'themeStyleOverrides'],
   }));
 
   describe('prop: inputProps', () => {
     it('should be able to provide a custom classes property', () => {
-      const { container } = render(
-        <Select
-          inputProps={{
-            classes: { root: 'root' },
-          }}
-          value=""
-        />,
+      let container;
+
+      expect(() => {
+        const { container: innerContainer } = render(
+          <Select
+            inputProps={{
+              classes: { root: 'root' },
+            }}
+            value=""
+          />,
+        );
+        container = innerContainer;
+      }).toErrorDev(
+        'Material-UI: The key `root` provided to the classes prop is not implemented in Select.',
```

Comment:
It throws error as `classes` inside the `Select` is not empty object, as it is not created with `withStyles`

Reply:
I have done b244bbb57b103c41acb64bf2ea2f8fe01bc5796f as a proposal to solve the issue. The alternative would be to have our own mergeClasses helper. The one from styles is legacy not relevant anymore. The accepted classes are so dynamic now that we can't work for classes that don't exist (as far as I understand)